lib/active_fedora/rdf/indexing_service.rb

```diff
@@ -64,35 +64,107 @@ def resource
         object.resource
       end
 
-      def properties
-        object.class.properties
-      end
-
       def index_config
         object.class.index_config
       end
 
-      # returns a Hash, e.g.: {field => { values: [], type: :something, behaviors: [] }, ...}
+      # returns the field map instance
       def fields
-        field_map = {}.with_indifferent_access
-
-        index_config.each do |name, index_field_config|
-          type = index_field_config.data_type
-          behaviors = index_field_config.behaviors
-          next unless type and behaviors
-          next if kind_of_af_base?(name)
-          field_map[name] = { values: find_values(name), type: type, behaviors: behaviors}
+        field_map_class.new do |field_map| 
+          index_config.each { |name, index_field_config| field_map.insert(name, index_field_config, object) }
         end
-        field_map
       end
 
-      def kind_of_af_base?(name)
-        config = properties[name.to_s]
-        config && config[:class_name] && config[:class_name] < ActiveFedora::Base
+      # Override this method to use your own FieldMap class for custom indexing of objects and properties
+      def field_map_class
```

Comment:
I'm welcome to be overridden, but this is a textbook factory. `field_map_factory`?